system/Common.php

```diff
@@ -447,13 +447,11 @@ function force_https(int $duration = 31536000, RequestInterface $request = null,
 
 		$baseURL = config(App::class)->baseURL;
 
-		// If we already use 'https' then return immediately
 		if (strpos($baseURL, 'https://') === 0)
 		{
-			return;
+			$baseURL = (string) substr($baseURL, strlen('https://'));
 		}
-
-		if (strpos($baseURL, 'http://') === 0)
+		else if (strpos($baseURL, 'http://') === 0)
```

Comment:
One little change - just to be compliant with our code standard:
```suggestion
		elseif (strpos($baseURL, 'http://') === 0)
```